flink-runtime/src/main/java/org/apache/flink/runtime/webmonitor/RestfulGateway.java

```diff
@@ -137,4 +138,21 @@
 			JobID jobId, @RpcTimeout Time timeout) {
 		throw new UnsupportedOperationException();
 	}
+
+	/**
+	 * Triggers a savepoint with the given savepoint directory as a target.
+	 *
+	 * @param targetDirectory Target directory for the savepoint.
+	 * @param timeout for the asynchronous operation
+	 * @return A future to the completed checkpoint
+	 * @throws IllegalStateException If no savepoint directory has been
+	 *                               specified and no default savepoint directory has been
+	 *                               configured
+	 */
+	default CompletableFuture<CompletedCheckpoint> triggerSavepoint(
```

Comment:
I'm wondering whether we actually need to expose the `CompletedCheckpoint`. Wouldn't it be enough to return the external path here? That way we would avoid further coupling between external components and internal data structures.

Reply:
I can make it return the external pointer.

Reply:
Cool :-)